pkg/virt-controller/watch/vmi/vmi_test.go

```diff
@@ -3526,6 +3526,44 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 		})
 	})
 
+	Context("feature configurables conditions", func() {
+		It("should update the conditions if the downwardMetrics feature is disabled in a running VM", func() {
+			vmi := newPendingVirtualMachine("testvm")
+			vmi.Spec.Domain.Devices.DownwardMetrics = &virtv1.DownwardMetrics{}
+			vmi.Status.Phase = virtv1.Running
+			pod := newPodForVirtualMachine(vmi, k8sv1.PodRunning)
+			addPod(pod)
+			addVirtualMachine(vmi)
+
+			controller.Execute()
+
+			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(MatchFields(IgnoreExtras,
+				Fields{
+					"Type":    Equal(virtv1.VirtualMachineInstanceOutOfSyncConfiguration),
+					"Status":  Equal(k8sv1.ConditionFalse),
+					"Message": Equal("The DownwardMetrics feature is disabled but still in use"),
+				})),
+			)
+		})
+
+		It("should stay in Pending and add the conditions if downwardMetrics is requested but no enabled", func() {
+			vmi := newPendingVirtualMachine("testvm")
+			vmi.Spec.Domain.Devices.DownwardMetrics = &virtv1.DownwardMetrics{}
+			addVirtualMachine(vmi)
+
+			controller.Execute()
+
+			testutils.ExpectEvent(recorder, kvcontroller.FeatureNotEnabled)
+			expectVMIWithMatcherConditions(vmi.Namespace, vmi.Name, ContainElement(MatchFields(IgnoreExtras,
```

Comment:
I would expect that if the VM configuration is not valid that it would not create a VMI at all?

Reply:
Yes, but if the VM is requested to start, in case that the configuration is not valid, would the VM try to create the VMi object again? If this is not the case, the user will need to stop/start the VM again, whereas in this way if the feature is enabled, it will just start.

Reply:
Changed. The VMI object is not created now if the configuration is invalid.